### PR TITLE
[5.3] Create an abstract HTTP channel

### PR DESCRIPTION
**Pull request by lucasmichot**
Extract Slack HTTP to an abstract HTTP channel.
This allows to implement other HTTP-related driver (which are many)

### CODE REVIEW

File: src/Illuminate/Notifications/Channels/HttpChannel.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+     * Create a new HTTP channel instance.
+     *
+     * @param  \GuzzleHttp\Client  $http
+     * @return void
+     */
+    public function __construct(HttpClient $http)
+    {
+        $this->http = $http;
+    }
+}
```
Comment (GrahamCampbell) [inline src/Illuminate/Notifications/Channels/HttpChannel.php]: Maybe make an abstract send function?

Comment (lucasmichot) [inline src/Illuminate/Notifications/Channels/HttpChannel.php]: I feel a `send` function would belongs to an interface?
See #14668

Comment (GrahamCampbell) [inline src/Illuminate/Notifications/Channels/HttpChannel.php]: Right, but this abstract client doesn't implement anything.

Comment (lucasmichot) [inline src/Illuminate/Notifications/Channels/HttpChannel.php]: It makes more sense the extending classes to implement the interface
As this abstract class will only be extended by HTTP related Channels

Comment (GrahamCampbell) [inline src/Illuminate/Notifications/Channels/HttpChannel.php]: On that logic, I see this abstract class as pointless then. Makes no sense to have this if only for a constructor. Infact, I think this is is an antipattern.

Comment (lucasmichot) [inline src/Illuminate/Notifications/Channels/HttpChannel.php]: Let see if #14668 is merged first, then we can make HttpChannel implements the contract

**Comment by GrahamCampbell** [comment]
Needs rebasing. :P

**Comment by lucasmichot** [comment]
Done

**Comment by taylorotwell** [comment]
Going to hold off on this one for now.

**Pull request closed without merging**